### PR TITLE
Update dependency jest to v29.7.0

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:

* Update dependency `jest` to `v29.7.0`:
  - update to new [snapshot-format](https://jestjs.io/docs/upgrading-to-jest29#snapshot-format)
  - update to new [timer-mocks](https://jestjs.io/docs/next/timer-mocks) API

* Fixed problems in `monitor` package:
  - Fixed acceptance tests
  - Added package validations to Dockerfile

* Suppress `<Suspense>` warning when running frontend test

* Update base alpine image to `v3.20`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -46,7 +46,7 @@ dashboard:
         pull-request: ~
       steps:
         check:
-          image: 'node:20-alpine3.18'
+          image: 'node:20-alpine3.20'
```
Comment (petersutter) [inline .ci/pipeline_definitions:49]: hm, seems like the renovate custom manager does not work in this case 🤔  https://github.com/gardener/dashboard/blob/c279ce4a1bf0ef6b5f4a324137d407d01bc891ec/.github/renovate.json5#L18-L24

File: packages/monitor/__tests__/acceptance/api.metrics.test.js [hunk 2]
```diff
@@ -40,9 +40,9 @@ describe('api', () => {
       const res = await agent
         .get('/unknown')
         .expect(404)
-
+      console.log(res.body)
```
Comment (petersutter) [inline packages/monitor/__tests__/acceptance/api.metrics.test.js:43]: remove

File: .ci/pipeline_definitions (hunk 1 above)
Comment (holgerkoser) [reply .ci/pipeline_definitions:49]: 🤔

File: packages/monitor/__tests__/acceptance/api.metrics.test.js (hunk 2 above)
Comment (holgerkoser) [reply packages/monitor/__tests__/acceptance/api.metrics.test.js]: ✔️

File: .ci/pipeline_definitions (hunk 1 above)
Comment (holgerkoser) [reply .ci/pipeline_definitions:49]: There was also no update for the node image in the [Dockerfile](https://github.com/gardener/dashboard/blob/c279ce4a1bf0ef6b5f4a324137d407d01bc891ec/Dockerfile#L6)

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**